oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/KeycloakRBACAuthorizer.java

```diff
@@ -0,0 +1,436 @@
+            }
+        }
+        return enumScopes;
+    }
+
+    boolean delegateIfRequested(RequestChannel.Session session, Operation operation, Resource resource, JsonNode authz) {
+        if (delegateToKafkaACL) {
+            boolean granted = super.authorize(session, operation, resource);
+
+            boolean grantLogOn = granted && GRANT_LOG.isDebugEnabled();
+            boolean denyLogOn = !granted && DENY_LOG.isDebugEnabled();
+
+            if (grantLogOn || denyLogOn) {
+                String status = granted ? "GRANTED" : "DENIED";
+                String message = "Authorization " + status + " by ACL - user: " + session.principal() + ", operation: " + operation + ", resource: " + resource;
+
+                if (grantLogOn) {
+                    GRANT_LOG.debug(message);
+                } else if (denyLogOn) {
+                    DENY_LOG.debug(message);
+                }
+            }
+            return granted;
+        }
+
+        if (DENY_LOG.isDebugEnabled()) {
+            DENY_LOG.debug("Authorization DENIED - cluster: " + clusterName + ", user: " + session.principal() + ", operation: " + operation + ", resource: " + resource + "\n permissions: " + authz);
+        }
+        return false;
+    }
+
+    JsonNode fetchAuthorizationGrants(String token) {
+
+        String authorization = "Bearer " + token;
+
+        StringBuilder body = new StringBuilder("audience=").append(urlencode(clientId))
+                .append("&grant_type=").append(urlencode("urn:ietf:params:oauth:grant-type:uma-ticket"))
+                .append("&response_mode=permissions");
+
+        JsonNode response;
+
+        try {
+            response = post(tokenEndpointUrl, socketFactory, hostnameVerifier, authorization,
+                    "application/x-www-form-urlencoded", body.toString(), JsonNode.class);
+
+        } catch (HttpException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to fetch authorization data from authorization server: ", e);
+        }
+
+        return response;
+    }
+
+    @Override
+    public void addAcls(Set<Acl> acls, Resource resource) {
+        super.addAcls(acls, resource);
+    }
+
+    @Override
+    public boolean removeAcls(Set<Acl> aclsTobeRemoved, Resource resource) {
+        return super.removeAcls(aclsTobeRemoved, resource);
+    }
+
+    @Override
+    public boolean removeAcls(Resource resource) {
+        return super.removeAcls(resource);
+    }
+
+    @Override
+    public Set<Acl> getAcls(Resource resource) {
+        return super.getAcls(resource);
+    }
+
+    @Override
+    public scala.collection.immutable.Map<Resource, Set<Acl>> getAcls(KafkaPrincipal principal) {
+        return super.getAcls(principal);
+    }
+
+    @Override
+    public scala.collection.immutable.Map<Resource, Set<Acl>> getAcls() {
+        return super.getAcls();
+    }
```

Comment:
By delegating these to super you're effectively letting people use the Kafka Admin `createAcls()`, `alterAcls()` and `deleteAcls()` even when `delegateToKafkaACL==false`. That's a bit misleading. Maybe we shouldn't go as far as to throw some exception in that case, but I certainly think some WARN logging is in order.

Reply:
That's a fair point. Honestly, I don't quite know what to do with these. 
We could go with what you suggest, and throw exception if delegating to simple ACL is not enabled.

Reply:
Yeah, either exception or logging. @ppatierno @scholzj any preference?

Reply:
No strong opinion on my side, I am fine with both.

Reply:
The only use case I can see for allowing them to create, alter and delete ACLs if they're not using them would be if they were trying to, e.g. delete ACLs after moving to using authz services (or setting them up when moving the other way). That seems like quite a stretch.

Reply:
I made it so that exception is thrown if delegation to simple Kafka ACL isn't enabled.